plugin/trino-mariadb/src/main/java/io/trino/plugin/mariadb/MariaDbClientModule.java

```diff
@@ -37,6 +41,7 @@ public void configure(Binder binder)
     {
         binder.bind(JdbcClient.class).annotatedWith(ForBaseJdbc.class).to(MariaDbClient.class).in(Scopes.SINGLETON);
         binder.install(new DecimalModule());
+        newSetBinder(binder, ConnectorTableFunction.class).addBinding().toProvider(RemoteQuery.class).in(Scopes.SINGLETON);
```

Comment:
Why not to do it in `trino-base-jdbc`?

Reply:
it's not applicable to some connectors (clickhouse)